Distributions.cpp

```diff
@@ -0,0 +1,168 @@
+            bounding box
+ @return    _dist_location a vector that contains the coordinates of a point
+*/
+std::vector <double> sampleLocation(Boundaries bounds) {
+    std::vector <double> _dist_location;
+    double width;
+    double coord;
+    for (int axis=0; axis<3; ++axis) {
+        width = bounds.getSurfaceCoord(axis, 1)
+            - bounds.getSurfaceCoord(axis, 0);
+        coord = bounds.getSurfaceCoord(axis, 0) + width * urand();
+        _dist_location.push_back(coord);
+    }
+    return _dist_location;
+}
+
+/*
+ @brief     Function that samples the interaction type given an absorption
+            (0 = capture, 1 = fission)
+ @details   Based on cross-sections, the interaction is sampled as capture
+            (0) or fission (1).
+ @param     mat a Material object that contains information about the material
+ @param     group contains the energy group of the neutron
+ @return    An interaction type (0 = capture, 1 = fission)
+*/
+int sampleFission(Material* mat, int group) {
+    int fission = urand() < mat->getSigmaF(group) / mat->getSigmaA(group);
+    return fission;
+}
+
+/*
+ @brief     Samples the nunber of neutrons produced from a fission event
+ @param     mat a Material object that contains information about the material
+ @return    number of neutrons emitted from the sampled fission event
+*/
+int sampleNumFission(Material* mat) {
+    int lower = (int) mat->getNu();
+    int add = (int) (urand() < mat->getNu()-lower);
+    return lower + add;
+}
+
+/*
+ @brief     Samples a neutron position in the fission bank
+ @param     fission_bank a Fission object containing neutron locations
+ @return    sampled neutron location
+*/
+std::vector <double> sampleFissionSite(
+        std::vector <std::vector <double> > &fission_bank) {
+    int index = rand() % fission_bank.size();
+    return fission_bank[index];
+}
+
+/*
+ @brief     Samples an initial neutron energy group after fission
+ @param     chi the neutron emission spectrum from fission
+ @return    the group number of the emitted neutron
+*/
+int sampleNeutronEnergyGroup(std::vector <double> chi) {
+    double r = urand();
+    double chi_sum = 0.0;
+    for (int g=0; g<chi.size(); ++g) {
+        chi_sum += chi[g];
+        if (r<chi_sum) {
+            return g;
+        }
+    }
+    return chi.size() - 1;
+}
+
+/*
+ @brief     Samples the neutron energy group after a scattering event
+ @param     scattering_matrix the scattering cross section matrix
+ @param     group the neutron energy group before scattering
+ @return    the neutron group after scattering
+*/
+int sampleScatteredGroup(std::vector <double> &scattering_matrix, int group) {
+
+    // Get the total scattering cross-section from this group
+    int num_groups = scattering_matrix.size();
+    double scattering_total = 0;
+    for (int g=0; g < num_groups; ++g)
+        scattering_total += scattering_matrix[g];
+
+    // Sample the outgoing scattered energy group
```

Comment:
I notice sometimes you capitalize the beginning of comments (like here) and sometimes you don't. Either is fine but pick one style and use it throughout.